src/extension.ts

```diff
@@ -238,8 +238,8 @@ function removeOutputChannel( channelName: string) {
 	if (outputChannelMap.has(channelName)) {
 		podOutputChannel = outputChannelMap.get(channelName);
 		if (podOutputChannel !== undefined) {
-			podOutputChannel.dispose();
 			outputChannelMap.delete(channelName);
+			podOutputChannel.dispose();
```

Comment:
Attempted to change the order to see if it would remove the `Error: Channel has been closed` problem, but it doesn't. It makes sense to remove it and THEN dispose it however, so I left it in place.

Reply:
wondering if the dispose doesn't take care of the delete itself or the other way around and that causes the error message maybe.

Reply:
It doesn't seem to. Disposing the object doesn't remove it from the map so far as I can tell.

Reply:
Disposing the object doesn't seem to remove it from the map

Reply:
:/

Reply:
see https://github.com/jboss-fuse/vscode-atlasmap/blob/master/src/extension.ts#L139-L186

we have a similar logic there and never experienced issues? Isn't it the same kind of thing?

Reply:
will continue to chase this down in https://github.com/camel-tooling/vscode-camelk/issues/69